zilliqa/src/db.rs

```diff
@@ -474,6 +509,15 @@ impl Db {
                     "Invalid checkpoint file: account trie root hash mismatch: calculated {}, checkpoint file contained {}", hex::encode(account_trie.root_hash()?.as_slice()), hex::encode(account_trie_root)
                 ));
             }
+            if processed_storage_items > maximum_sql_parameters {
```

Comment:
I don't think the maximum SQL parameters matters here. The reason we're occasionally flushing to disk is to limit the memory usage of `mem_storage`? So can we instead just pick an arbitrary large-ish number for the maximum number of kv pairs we keep in memory at once?

Reply:
Ah yeah that's a fair point :+1: